MCU_Aurix/bootloader/src/isotp.c

```diff
@@ -205,18 +236,22 @@ uint8_t* isotp_rcv(uint32_t* total_length){
  */
 void process_can_to_isotp(uint32_t* rxData, IfxCan_DataLengthCode dlc){
 
-    //TODO: Discuss with sebastian that we would like to set the dlc for all messages.
+    if(dlc <= 0) // Need to assume that the data is also empty, using FBL_NEGATIVE_RESPONSE instead
+        uds_neg_response(FBL_NEGATIVE_RESPONSE, FBL_RC_INCORRECT_MSG_LEN_OR_INV_FORMAT);
 
     if(MAX_ISOTP_MESSAGE_LEN - (iso_RX->write_ptr - iso_RX->data) < dlc || iso_RX->ready_to_read != 0){
 
-        // TODO: send error message that we cannot receive further can messages.
+        // Request with negative response if case of error
+        if(rx_is_single_Frame(rxData, dlc, MAX_FRAME_LEN_CAN))
+            uds_neg_response(rxData[1], FBL_RC_BUSY_REPEAT_REQUEST);
```

Comment:
hier ist rxData[1] ein uint32_t und in uds_neg_response() wird aber ein uint8_t erwartet. Da kann was an daten verloren gehen.

Reply:
stimmt, es gibt sowieso darunter bereits einen uint8_t* data_ptr. Diesen habe ich verschoben und entsprechend angepasst.